File: solutions/java/log-levels/2/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
       //  return logLine.split(":")[1].trim();
        int split = logLine.indexOf(':') + 1;
        return logLine.substring(split).trim();
    }

    public static String logLevel(String logLine) {
        return logLine.split(":")[0].replace("[", "").replace("]", "").toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")"; 
    }
}

File: solutions/cpp/log-levels/2/log_levels.cpp
#include <string>
#include <iostream>

namespace log_line {
    std::string message(std::string line) {
        // return the message
        int cut = line.find(":");
        
        std::string ans = line.substr(cut + 2);
        
        return ans;
    }

    std::string log_level(std::string line) {
        // return the log level
        int cut = line.find(":");
        std::string ans = line.substr(1, cut-2);
        
        return ans;
    }

    std::string reformat(std::string line) {
        // return the reformatted message
        std::string ans = message(line) + " (" + log_level(line) + ")";
        
        return ans;
    }
}
